cli/validate_inputs.go

```diff
@@ -63,12 +63,14 @@ func validateTerragruntInputs(terragruntOptions *options.TerragruntOptions, work
 		}
 		terragruntOptions.Logger.Error("")
 	} else {
-		terragruntOptions.Logger.Info("All required inputs are passed in by terragrunt.")
+		terragruntOptions.Logger.Info(fmt.Sprintf("All required inputs are passed in by terragrunt. Strict mode enabled: %t", terragruntOptions.Strict))
 	}
 
-	// Return an error when there are misaligned inputs.
-	if len(unusedVars) > 0 || len(missingVars) > 0 {
-		return fmt.Errorf("Terragrunt configuration has misaligned inputs")
+	// Return an error when there are misaligned inputs. Terragrunt strict mode defaults to false. When it is false,
+	// an error will only be returned if required inputs are missing. When strict mode is true, an error will be
+	// returned if required inputs are missing OR if any unused variables are passed
+	if len(missingVars) > 0 || len(unusedVars) > 0 && terragruntOptions.Strict {
+		return fmt.Errorf("Terragrunt configuration has misaligned inputs. Strict mode enabled: %t", terragruntOptions.Strict)
 	}
```

Comment:
NIT: in non strict mode, it should emit a warning log at the end for unused variables.

Reply:
Is this separate from the warning message that is logged on line 50?

Reply:
Yea. The idea here is that most people will/might only look at the last message, so we want to give the overall summary at the end.

Reply:
Something like `"Terragrunt configuration has misaligned inputs, but running in relaxed mode so ignoring."`

Reply:
Ah roger, thanks :+1:

Reply:
Done in 03bd7fb